core/base/inc/TUrl.h

```diff
@@ -56,7 +56,7 @@ class TUrl : public TObject {
    enum EStatusBits { kUrlWithDefaultPort = BIT(14), kUrlHasDefaultPort = BIT(15) };
 
 public:
-   TUrl() = default;
+   TUrl() {} // NOLINT: not allowed to use = default because of TObject::kIsOnHeap detection, see ROOT-10300
```

Comment:
I don't understand that. Even `= default` *must* call the `TObject` constructor.

Reply:
> I don't understand that. Even = default must call the TObject constructor.

It calls constructor, but before it set all members to 0

Reply:
I see, so you're working around the compiler using it's freedom of us doing UB. It's not *really* the `=default` that forces this to happen; in principle the compiler could just do that for *any* object before calling its constructor...

So I think we're generally toast here. But I now understand how removing `=default` can avoid triggering this compiler optimization. Thanks for the explanation, @linev !